sympy/ntheory/generate.py

```diff
@@ -381,30 +383,34 @@ def prime(nth):
         .. [2] https://en.wikipedia.org/wiki/Prime_number_theorem#Approximations_for_the_nth_prime_number
         .. [3] https://en.wikipedia.org/wiki/Skewes%27_number
     """
-    n = as_int(nth)
-    if n < 1:
-        raise ValueError("nth must be a positive integer; prime(1) == 2")
-    if n <= len(sieve._list):
-        return sieve[n]
-
-    from sympy.functions.special.error_functions import li
-    from sympy.functions.elementary.exponential import log
-
-    a = 2 # Lower bound for binary search
-    b = int(n*(log(n) + log(log(n)))) # Upper bound for the search.
-
-    while a < b:
-        mid = (a + b) >> 1
-        if li(mid) > n:
-            b = mid
-        else:
-            a = mid + 1
-    n_primes = primepi(a - 1)
-    while n_primes < n:
-        if isprime(a):
-            n_primes += 1
-        a += 1
-    return a - 1
+    @classmethod
+    def eval(cls, nth):
+        if(not isinstance(nth, Integer)):
```

Comment:
You don't need parentheses here.

Reply:
Thanks for pointing. I will make it correct. What do you think about the change in the behavior of these functions with this new change?